tests/test_put_file.py

```diff
@@ -56,3 +62,58 @@ def test_no_change_postcommit(
     )
     assert commits.results[0] == latest_commit
     # in particular, this test asserts that no API exception happens in postcommit.
+
+
+def test_implicit_branch_creation(
+    random_file_factory: RandomFileFactory, fs: LakeFSFileSystem, repository: str, temp_branch: str
+) -> None:
+    # Create Changes to commit.
+    random_file = random_file_factory.make()
+    lpath = str(random_file)
+
+    with fs.scope(create_branch_ok=True):
+        # Pushing to an existing branch
+        rpath = f"{repository}/{temp_branch}/{random_file.name}"
+        fs.put(lpath, rpath)
+        commits = fs.client.commits_api.log_branch_commits(
+            repository=repository,
+            branch=temp_branch,
+        )
+        latest_commit = commits.results[0]  # commit log is ordered branch-tip-first
+        assert latest_commit.message == f"Add file {random_file.name}"
+
+        # Creates non-existing branch and then pushes to it.
+        non_existing_branch = "non-existing-" + "".join(random.choices(string.digits, k=8))
+        rpath = f"{repository}/{non_existing_branch}/{random_file.name}"
+        fs.put(lpath, rpath)
+        commits = fs.client.commits_api.log_branch_commits(
+            repository=repository,
+            branch=non_existing_branch,
+        )
```

Comment:
Please use this instead (`log_branch_commits` is deprecated in favor of `refs_api.log_commits`):

```suggestion
        commits = fs.client.refs_api.log_commits(
            repository=repository,
            ref=non_existing_branch,
        )
```